src/Tokenizer/Analyzer/Analysis/ArgumentAnalysis.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Analyzer\Analysis;
+
+/**
+ * @internal
+ */
+final class ArgumentAnalysis
+{
+    /**
+     * The default value of the argument.
+     *
+     * @var null|string
+     */
+    private $default;
+
+    /**
+     * The name of the argument.
+     *
+     * @var string
+     */
+    private $name;
+
+    /**
+     * The index where the name is located in the supplied Tokens object.
+     *
+     * @var int
+     */
+    private $nameIndex;
+
+    /**
+     * The type of the argument.
+     *
+     * @var null|string
+     */
+    private $type;
+
+    /**
+     * @var null|int
+     */
+    private $typeIndexStart;
+
+    /**
+     * @var null|int
+     */
+    private $typeIndexEnd;
+
+    /**
+     * ArgumentAnalysis constructor.
+     *
+     * @param string      $name
+     * @param int         $nameIndex
+     * @param null|string $default
+     * @param null|string $type
+     * @param null|int    $typeIndexStart
+     * @param null|int    $typeIndexEnd
+     */
+    public function __construct($name, $nameIndex, $default, $type, $typeIndexStart, $typeIndexEnd)
+    {
+        $this->name = (string) $name;
```

Comment:
what is the valid case of passing here non-string and cast it into string?
i wouldn't support that, we expected a string, when we will bump requirements to 7.0 we will have string typehint in constructor, making compiler crash on non-strings and no casing will be done ever.

please, remove casing or proof non-string type is valid input, but then change the typehinting in docblock.

same in all other places (not only this class).

-------------------------------

i do know that casing is kind-of defensive approach, but if you want to go defensive, raise InvalidArgumentException if passed non-php typehinted argument is violating typehint from docblock

Reply:
Indeed: I've added it as a defensive approuch since typehints are not supported yet. Will remove it ...